anitya/lib/backends/gitlab.py

```diff
@@ -101,7 +101,13 @@ def get_versions(cls, project):
 
         tags = []
         for tag in json:
-            tags.append(tag["name"])
+            name = tag["name"]
```

Comment:
The version API (used by the ``Project.versions`` property) should handle both sorting and stripping prefixes. If the UI isn't displaying correctly-stripped versions, it should use that property to get the list, I think.

Reply:
I don't think the version API is striping the prefix, because I saw this in few backends already.

Reply:
See https://github.com/release-monitoring/anitya/blob/4ebc10295b985d78770ce068f92f2164802ef822/anitya/lib/versions/base.py#L45

Reply:
In this case than something is wrong with this. I need to check if this works or what happened.

Reply:
Ok, this issue is more complex than I thought. When we are doing check for new versions, we are calling get_versions directly on backend. This works on most of the backends, because they are doing the prefix handling.
I need to refactor this a little to make it working. Probably applying project prefix on retrieved versions before comparing them with existing versions.

Reply:
I found out why this issue wasn't found before. Because every backend used regex to obtain new versions and function `get_version_by_regex` (see https://github.com/release-monitoring/anitya/blob/master/anitya/lib/backends/__init__.py#L284) is erasing the prefix. Github and Gitlab are using APIs instead of regexes.
I think the regex shouldn't be erased here, but rather in `check_project_release` or it shoudn't be removed at all and left this on the `parse` method in version plugin.

There are two options right now:
1) We could remove the prefix before saving the version to database - best place is `check_project_release`
2) We shouldn't remove the prefix at all - but this could cause issue, if the prefix is changed in the future
2) We